Tetragrama/src/Components/LogUIComponent.cpp

```diff
@@ -26,11 +28,47 @@ namespace Tetragrama::Components {
         ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
 
         if (m_content.size() > 0) {
+
             const char* buf     = m_content.begin();
             const char* buf_end = m_content.end();
-            ImGui::TextUnformatted(buf, buf_end);
-        }
 
+
+            ImGuiListClipper clipper;
+            clipper.Begin(m_content_line_offsets.Size);
+            while (clipper.Step()) {
+                for (int line_no = clipper.DisplayStart; line_no < clipper.DisplayEnd; line_no++) {
+                    const char* line_start = buf + m_content_line_offsets[line_no];
+                    const char* line_end   = (line_no + 1 < m_content_line_offsets.Size) ? (buf + m_content_line_offsets[line_no + 1] - 1) : buf_end;
+                   
+                    ImVec4 color;
```

Comment:
We don't need those `color `and `has_color `variables

Reply:
okey

Reply:
done